### ISSUE TITLE
Incorrect behaviour using resnet18v1/resnet50v1 and onnxruntime: extremely low accuracy.

### ISSUE THREAD

**Issue by yqtianust** [microsoft/onnxruntime#1958]
**Describe the bug**
Run resnet18v1 using onnxruntime (python) get incorrect output.

 Inferencing using resnet18v1 (from onnx model zoo [resnet18v1](https://github.com/onnx/models/tree/master/vision/classification/resnet) and onnxruntime get incorrect output. For example, all 50 inputs from 1st class of imagenet have same top-1 prediction: 932. The overall accuracy is 50/50000

If I switch to resnet18v2, the accuracy is correct (same as the the document).

If I use MXNet as backend, the accuracy is also correct. 

**Urgency**
None

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- ONNX Runtime installed from (source or binary):binary
- ONNX Runtime version: 0.5.0
- Python version: 3.6
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.0
- GPU model and memory: RTX 2080Ti, 12GB

**To Reproduce**
`
model_path = './resnet18v1.onnx'
m = rt.InferenceSession(model_path)
input_node = m.get_inputs()[0].name
output_node = m.get_outputs()[0].name
for data, label in yield_single():
	result = m.run([output_node], {input_node:data})  
	top_1 = postprocess(result[0])
`
There should be no problem on the other functions since once I changed to resnet18v2 or resnet50v2, the accuracy back to normal.

**Expected behavior**
acceptable accuracy

**Comment by snnn** [comment]
You may try the resnet models converted from Tensorflow, they are all correct. 

https://github.com/tensorflow/models/tree/master/research/slim
https://github.com/onnx/tensorflow-onnx/tree/master/tf2onnx

**Comment by snnn** [comment]
see https://github.com/onnx/models/issues/156

**Comment by yqtianust** [comment]
Thanks @snnn . I will try the model you mentioned.

I read your second reply. However, I don't understand why resent50v2 can work. I noticed that the difference between v1 and v2 is 

> ResNet v2 uses pre-activation function whereas ResNet v1 uses post-activation for the residual blocks.

Why the issues mentioned in  #onnx/models#156 will has such influence here?

**Comment by snnn** [comment]
#2092

**Comment by snnn** [comment]
mxnet_arcface:Non-zero status code returned while running BatchNormalization node. Name:'stage1_unit1_bn1' Status Message: Invalid input scale: NumDimensions() != 3

We still can't run this model

**Comment by snnn** [comment]
It's not a bug of onnxruntime, it's a bug of the mxnet converter. 
When you open the model, you'll see the second node of the model is BatchNormalization. It has an attribute with name of  "spatial" and value of 0.  But based on the input shapes the value should be **one** here. 

Please see:
https://github.com/onnx/onnx/blob/master/docs/Changelog.md#BatchNormalization-7

For example, for the "scale" input, the spec says: If spatial is true, the dimension of scale should be (C). If spatial is false, the dimensions of scale should be (C x D1 x ... x Dn)

In this case, the scale input has type of float32 and shape of [64]. So the spatial should be true.

As it's not an onnxruntime bug, I'll close this issue now.  Please report the bug to onnx model zoo. 
https://github.com/onnx/models/issues

Thank you!
